portable/GCC/ARM_CM3_MPU/port.c

```diff
@@ -787,6 +795,32 @@ static void prvRestoreContextOfFirstTask( void )
  */
 BaseType_t xPortStartScheduler( void )
 {
+    /* Applications that route program control to the FreeRTOS interrupt
+     * handlers through intermediate handlers (indirect routing) should set
+     * configCHECK_HANDLER_INSTALLATION to 0 in FreeRTOSConfig.h. Direct
+     * routing, which is validated here when configCHECK_HANDLER_INSTALLATION
+     * is 1, is preferred when possible. */
+    #if ( configCHECK_HANDLER_INSTALLATION == 1 )
+    {
+        const portISR_t * const pxVectorTable = portSCB_VTOR_REG;
+
+        /* Verify correct installation of the FreeRTOS handlers for SVCall and
+         * PendSV. Do not check the installation of the SysTick handler because
+         * the application may provide the OS tick without using the SysTick
+         * timer by overriding the weak function vPortSetupTimerInterrupt().
+         *
+         * Assertion failures here can be caused by incorrect installation of
+         * the FreeRTOS handlers. For help installing the handlers, see
+         * https://www.FreeRTOS.org/FAQHelp.html
+         *
+         * Systems with a configurable address for the interrupt vector table
+         * can also encounter assertion failures or even system faults here if
+         * VTOR is not set correctly to point to the application's vector table. */
+        configASSERT( pxVectorTable[ portVECTOR_INDEX_SVC ] == vPortSVCHandler );
+        configASSERT( pxVectorTable[ portVECTOR_INDEX_PENDSV ] == xPortPendSVHandler );
+    }
+    #endif /* configCHECK_HANDLER_INSTALLATION */
+
```

Comment:
The MPU ports have a different mechanism for setting the priority of SVCall.  It needs improvement, probably outside the scope of this PR.  See [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/a8650b99a33cd6fbc557748ef719e0229296676d/portable/GCC/ARM_CM4_MPU/port.c#L454-L456).

Reply:
Thanks for the link. Looking further into this, In Armv7-M MPU ports SVC handler is configured with priority `configMAX_SYSCALL_INTERRUPT_PRIORITY - 1`. Wouldn't this leads to priority escalation since there can be other exceptions/interrupts that can be of higher priority than SVCall?
From Armv7-M architecture reference manual:
```
Priority escalation
When the current execution priority is less than HardFault, the processor escalates the exception priority to
HardFault in the following cases:
• When the group priority of a pending synchronous fault or Supervisor Call is lower than or equal to the
currently executing priority, inhibiting normal preemption. This applies to all synchronous exceptions, both
faults and SVCalls. This includes a DebugMonitor exception caused by executing a BKPT instruction, but
excludes all other DebugMonitor exceptions.
```

@jefftenney @aggarg Any thoughts on this? I'm happy to start a FreeRTOS forum thread for this discussion.

What I'm trying to make sure that, we achieve same behaviour for both Armv7-M MPU and non-MPU ports. If setting `configMAX_SYSCALL_INTERRUPT_PRIORITY - 1` is needed for MPU ports, then we should set SVC handler priority to `configMAX_SYSCALL_INTERRUPT_PRIORITY - 1` for non-MPU port as well. Currently this PR sets it to 0 (which is inline with previous behaviour as SHPR2 register is set to 0x0 during reset).

Reply:
@urutva If you don't mind including that clean-up in this PR, I would be happy to clean it up.  Remember that `configMAX_SYSCALL_INTERRUPT_PRIORITY - 1` is a *higher* priority than `configMAX_SYSCALL_INTERRUPT_PRIORITY`.

To clean it up, I would make the ARMv7-M MPU ports use SVC priority of zero just like all the other Cortex M ports do.  I'm happy to do it in this PR.